tensorflow_addons/metrics/f1_test.py

```diff
@@ -0,0 +1,125 @@
+        f1_obj1 = F1Score(num_classes=3, average='macro')
+        f1_obj2 = F1Score(num_classes=3, average='weighted')
+
+        self.evaluate(tf.compat.v1.variables_initializer(f1_obj.variables))
+        self.evaluate(tf.compat.v1.variables_initializer(f1_obj1.variables))
+        self.evaluate(tf.compat.v1.variables_initializer(f1_obj2.variables))
+        return f1_obj, f1_obj1, f1_obj2
+
+    def initialize_vars_none(self):
+        f1_none = F1Score(num_classes=3, average=None)
+
+        self.evaluate(tf.compat.v1.variables_initializer(f1_none.variables))
+        return f1_none
+
+    def update_obj_states(self, f1_obj, f1_obj1, f1_obj2, actuals, preds):
+        update_op1 = f1_obj.update_state(actuals, preds)
+        update_op2 = f1_obj1.update_state(actuals, preds)
+        update_op3 = f1_obj2.update_state(actuals, preds)
+        self.evaluate(update_op1)
+        self.evaluate(update_op2)
+        self.evaluate(update_op3)
+
+    def update_obj_states_none(self, f1_none, actuals, preds):
+        update_op1_none = f1_none.update_state(actuals, preds)
+        self.evaluate(update_op1_none)
+
+    def check_results(self, obj, value):
+        self.assertAllClose(value, self.evaluate(obj.result()), atol=1e-5)
+
+    def test_f1_perfect_score(self):
+        actuals = tf.constant([[1, 1, 1], [1, 0, 0], [1, 1, 0]],
+                              dtype=tf.int32)
+        preds = tf.constant([[1, 1, 1], [1, 0, 0], [1, 1, 0]], dtype=tf.int32)
+        # Initialize
+        f1_obj, f1_obj1, f1_obj2 = self.initialize_vars()
+        # Update
+        self.update_obj_states(f1_obj, f1_obj1, f1_obj2, actuals, preds)
+        # Check results
+        self.check_results(f1_obj, 1.0)
+        self.check_results(f1_obj1, 1.0)
+        self.check_results(f1_obj2, 1.0)
+
+    def test_f1_worst_score(self):
+        actuals = tf.constant([[1, 1, 1], [1, 0, 0], [1, 1, 0]],
+                              dtype=tf.int32)
+        preds = tf.constant([[0, 0, 0], [0, 1, 0], [0, 0, 1]], dtype=tf.int32)
+        # Initialize
+        f1_obj, f1_obj1, f1_obj2 = self.initialize_vars()
+        # Update
+        self.update_obj_states(f1_obj, f1_obj1, f1_obj2, actuals, preds)
+        # Check results
+        self.check_results(f1_obj, 0.0)
+        self.check_results(f1_obj1, 0.0)
+        self.check_results(f1_obj2, 0.0)
+
+    def test_f1_random_score(self):
+        actuals = tf.constant([[1, 1, 1], [1, 0, 0], [1, 1, 0]],
+                              dtype=tf.int32)
+        preds = tf.constant([[0, 0, 1], [1, 1, 0], [1, 1, 1]], dtype=tf.int32)
+        # Initialize
+        f1_obj, f1_obj1, f1_obj2 = self.initialize_vars()
+        # Update
+        self.update_obj_states(f1_obj, f1_obj1, f1_obj2, actuals, preds)
+        # Check results
+        self.check_results(f1_obj, 0.6666666)
+        self.check_results(f1_obj1, 0.6555555)
+        self.check_results(f1_obj2, 0.6777777)
+
+    def test_f1_none_score(self):
```

Comment:
Rather make making a separate test case for `average=None`, include it in the other `test_f1_*` methods. So we can check for `perfect`, `random` and `worst` for the `None` case.